### PR TITLE
docstore: store revision into doc on writes

### PR DESCRIPTION
**Pull request by jba**
- All providers' writes store the new revision into the argument doc.

- Add conformance tests to check that.

### CODE REVIEW

File: internal/docstore/docstore.go [hunk 1]
```diff
@@ -107,19 +107,30 @@ func (l *ActionList) add(a *Action) *ActionList {
...
-// Create adds an action that creates a new document to the given ActionList, and returns the ActionList.
-// The document must not already exist; an error for which gcerrors.Code returns
-// AlreadyExists is returned if it does. If the document doesn't have key fields, it
-// will be given key fields with unique values.
+// Create adds an action that creates a new document to the given ActionList, and
+// returns the ActionList. The document must not already exist; an error with code
+// AlreadyExists is returned if it does. (See gocloud.dev/gcerrors for more on error
+// codes.)
+//
+// If the document doesn't have key
```
Comment (shantuo) [inline internal/docstore/docstore.go]: nit: this line seems too short.

File: internal/docstore/firedocstore/fs.go [hunk 2]
```diff
@@ -580,16 +580,14 @@ func (c *collection) doCommitCall(ctx context.Context, call *commitCall, errs []
...
-	// 		// Ignore errors. It's fine if the doc doesn't have a revision field.
-	// 		_ = call.actions[j].Doc.SetField(docstore.RevisionField, wr.UpdateTime)
-	// 		j++
-	// 	}
-	// }
+	j := 0
+	for i, wr := range wrs {
+		if _, ok := call.writes[i].Operation.(*pb.Write_Transform); !ok {
+			// Ignore errors. It's fine if the doc doesn't have a revision field.
+			_ = call.actions[j].Doc.SetField(docstore.RevisionField, wr.UpdateTime)
```
Comment (shantuo) [inline internal/docstore/firedocstore/fs.go]: nit: delete `_ = `

File: internal/docstore/memdocstore/mem.go [hunk 3]
```diff
@@ -223,6 +223,7 @@ func (c *collection) runAction(ctx context.Context, a *driver.Action) error {
 			return err
 		}
 		c.changeRevision(doc)
+		_ = a.Doc.SetField(c.opts.RevisionField, doc[c.opts.RevisionField])
```
Comment (shantuo) [inline internal/docstore/memdocstore/mem.go]: ditto

File: internal/docstore/docstore.go (hunk 1 above)
Comment (jba) [reply internal/docstore/docstore.go]: will reformat in a later CL

Comment (jba) [reply internal/docstore/docstore.go]: nvm, reformatted.

File: internal/docstore/firedocstore/fs.go (hunk 2 above)
Comment (jba) [reply internal/docstore/firedocstore/fs.go]: done

**Pull request merged**